src/Catalyst.Abstractions/Kvm/Models/BlockParameters.cs

```diff
@@ -22,63 +22,98 @@
+        {
+            Type = BlockParameterType.BlockHash;
+            BlockHash = blockHash;
+            RequireCanonical = requireCanonical;
+        }
+        
+        public static BlockParameter FromJson(string jsonValue)
         {
             switch (jsonValue)
             {
                 case { } earliest when string.Equals(earliest, "earliest", StringComparison.InvariantCultureIgnoreCase):
-                    Type = BlockParameterType.Earliest;
-                    return;
+                    return Earliest;
                 case { } pending when string.Equals(pending, "pending", StringComparison.InvariantCultureIgnoreCase):
-                    Type = BlockParameterType.Pending;
-                    return;
+                    return Pending;
                 case { } latest when string.Equals(latest, "latest", StringComparison.InvariantCultureIgnoreCase):
-                    Type = BlockParameterType.Latest;
-                    return;
+                    return Latest;
                 case { } empty when string.IsNullOrWhiteSpace(empty):
-                    Type = BlockParameterType.Latest;
-                    return;
+                    return Latest;
                 case null:
-                    Type = BlockParameterType.Latest;
-                    return;
+                    return Latest;
+                case { } hash when hash.Length == 66 && hash.StartsWith("0x"):
+                    return Latest;
                 default:
-                    Type = BlockParameterType.BlockNumber;
-                    BlockNumber = LongConverter.FromString(jsonValue.Trim('"'));
-                    return;
+                    return new BlockParameter(LongConverter.FromString(jsonValue.Trim('"')));
             }
         }
 
         public override string ToString()
         {
-            return $"{Type}, {BlockNumber}";
+            return $"{Type}, {BlockNumber?.ToString() ?? BlockHash?.ToString()}";
+        }
+        
+        public bool Equals(BlockParameter other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return Type == other.Type && BlockNumber == other.BlockNumber && BlockHash == other.BlockHash && other.RequireCanonical == RequireCanonical;
         }
 
-        public FilterBlock ToFilterBlock() =>
-            BlockNumber != null
-                ? new FilterBlock(BlockNumber ?? 0)
-                : new FilterBlock(Type.ToFilterBlockType());
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((BlockParameter) obj);
+        }
+
+        public override int GetHashCode()
+        {
+            throw new NotSupportedException();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this 'throw' statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979613)